src/adapter/src/coord/command_handler.rs

```diff
@@ -400,10 +400,11 @@ impl Coordinator {
         };
 
         let session_type = metrics::session_type_label_value(ctx.session().user());
+        let application_name = ctx.session().application_name();
```

Comment:
This could have a high cardinality. Could we use the canonicalized version we have that limits itself to a known set?

Reply:
If we're likely to be using this for investigation, would that not filter out a lot of useful information? If we are careful to only add this label to the `query_total` metric, it may be a worthwhile tradeoff?

Reply:
That's fine with me if cloud people also think it's ok. I don't expect the cardinality to be more than some dozens per customer. For reference, our known set is at https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/session/hint.rs

Reply:
If the expectation is that it's dozens rather than hundreds, and we're careful only to add the label to one or two metrics, IMHO the additional debugging help is worth it. Could we add a warning comment to that effect, though?

Reply:
This all makes sense to me, thanks @mjibson and @necaris! I added a comment where we define our metrics describing the problem, let me know if I should add it anywhere else